### PR TITLE
ollama: fix darwin build failure by changing dependency version

### LINKED ISSUE #345333: Build failure: ollama (aarch64-darwin, sandbox)
### Steps To Reproduce

```
$ nix build github:nixos/nixpkgs/4d2f662676c79b6e30ec9e8f7a41236d5f883687#ollama
```

### Build log

https://gist.github.com/n8henrie/0b2581aa8ede32fd2e4141c5d2473f66

### Notify maintainers

@abysssol
@dit7ya
@elohmeier
@roydubnium

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.0.0, macOS 15.0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(n8henrie): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/x2qbrd60nz1y86ha15k7mkc2i1d7xb37-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by abysssol**
## Description of changes

fixes  #345333

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ol/ollama/package.nix [hunk 1]
```diff
@@ -93,7 +93,7 @@ let
     ];
   };
 
-  metalFrameworks = with darwin.apple_sdk_11_0.frameworks; [
+  metalFrameworks = with darwin.apple_sdk.frameworks; [
```
Comment (winterqt) [inline pkgs/by-name/ol/ollama/package.nix:96]: This is a noop on aarch64-darwin, therefore is not actually the fix.

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ol/ollama/package.nix (hunk 1 above)
Comment (abysssol) [reply pkgs/by-name/ol/ollama/package.nix:96]: Thanks. You seem to know more about darwin than I do: do you know what the "right" package set to use would be? Version 11.0 seems not to be the best choice if there are other versions (such as 11.3) in use on real systems.

Comment (winterqt) [reply pkgs/by-name/ol/ollama/package.nix:96]: 11.0 is the best we have at the moment. (It'll be bumped to 11.3 soon, but that's just a minor release of 11.3.)

Comment (winterqt) [reply pkgs/by-name/ol/ollama/package.nix:96]: (I'd be surprised if anyone is still running Big Sur in 2024? Maybe I'm wrong. 😅)

**Pull request closed without merging**